client/consensus/src/tests.rs

```diff
@@ -0,0 +1,775 @@
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Tanssi is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Tanssi.  If not, see <http://www.gnu.org/licenses/>.
+
+// This tests have been greatly influenced by
+// https://github.com/paritytech/substrate/blob/master/client/consensus/aura/src/lib.rs#L832
+// Most of the items hereby added are intended to make it work with our current consensus mechanism
+
+#[cfg(test)]
+mod tests {
+    use {
+        crate::{
+            consensus_orchestrator::{
+                build_orchestrator_aura_worker, BuildOrchestratorAuraWorkerParams,
+            },
+            InherentDataProviderExt, LOG_TARGET,
+        },
+        cumulus_client_consensus_common::ParachainConsensus,
+        cumulus_primitives_core::PersistedValidationData,
+        futures::prelude::*,
+        futures_timer::Delay,
+        nimbus_primitives::{
+            CompatibleDigestItem, NimbusId, NimbusPair, NIMBUS_ENGINE_ID, NIMBUS_KEY_ID,
+        },
+        parking_lot::Mutex,
+        sc_block_builder::BlockBuilderProvider,
+        sc_client_api::{BlockchainEvents, HeaderBackend},
+        sc_consensus::{BoxJustificationImport, ForkChoiceStrategy},
+        sc_consensus_aura::SlotProportion,
+        sc_consensus_slots::{BackoffAuthoringOnFinalizedHeadLagging, SimpleSlotWorker, SlotInfo},
+        sc_keystore::LocalKeystore,
+        sc_network_test::{Block as TestBlock, *},
+        sp_consensus::{
+            BlockOrigin, EnableProofRecording, Environment, NoNetwork as DummyOracle, Proposal,
+            Proposer, SelectChain, SyncOracle,
+        },
+        sp_consensus_aura::{inherents::InherentDataProvider, SlotDuration},
+        sp_consensus_slots::Slot,
+        sp_core::{
+            crypto::{ByteArray, Pair},
+            H256,
+        },
+        sp_inherents::{CreateInherentDataProviders, InherentData},
+        sp_keyring::sr25519::Keyring,
+        sp_keystore::Keystore,
+        sp_runtime::{
+            traits::{Block as BlockT, Header as _},
+            Digest, DigestItem,
+        },
+        sp_timestamp::Timestamp,
+        std::{sync::Arc, task::Poll, time::Duration},
+        substrate_test_runtime_client::TestClient,
+    };
+
+    // Duration of slot time
+    const SLOT_DURATION_MS: u64 = 1000;
+
+    type Error = sp_blockchain::Error;
+
+    struct DummyFactory(Arc<TestClient>);
+    // We are going to create API because we need this to test runtime apis
+    // We use the client normally, bnut for testing certain runtime-api calls,
+    // we basically mock the runtime-api calñs
```

Comment:
```suggestion
    // We are going to create API because we need this to test runtime apis
    // We use the client normally, but for testing certain runtime-api calls,
    // we basically mock the runtime-api calls
```